mso/resource_mso_schema.go

```diff
@@ -60,26 +66,99 @@ func resourceMSOSchema() *schema.Resource {
 							Required:     true,
 							ValidateFunc: validation.StringLenBetween(1, 1000),
 						},
+						"description": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
 						"tenant_id": &schema.Schema{
 							Type:         schema.TypeString,
 							Required:     true,
 							ValidateFunc: validation.StringLenBetween(1, 1000),
 						},
+						"template_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
```

Comment:
Why make it required?

Reply:
Reasoning was of actively pushing user to select the correct type. Besides that I did not want to rely on potential changes to NDO api to impact the behaviour of template type when not providing it. 

If required not to be used I can change this to optional computed, or optional with default in the provided. Which one do you prefer? The optional computed setting might require a mso-go-client change since the value is now always set and else would probable need to be excluded from payload send.

Reply:
I assumed you would likely choose the mso-go-client option so decided to already prepare and test those changes. See PR https://github.com/ciscoecosystem/mso-go-client/pull/120 for mso-go-client changes. And this PR is updated also. Can reset the changes if decided otherwise. 

Would still need to update the mso-go-client dependency after release.